agent.py

```diff
@@ -90,6 +115,38 @@ def learn(self, mem):
       m.view(-1).index_add_(0, (u + offset).view(-1), (pns_a * (b - l.float())).view(-1))  # m_u = m_u + p(s_t+n, a*)(b - l)
 
     loss = -torch.sum(m * log_ps_a, 1)  # Cross-entropy loss (minimises DKL(m||p(s_t, a_t)))
+
+    with torch.no_grad():
```

Comment:
Can you explain why this extra no grad forward pass is being done? the forward pass above has gradients.